test/shared/src/main/scala/zio/test/package.scala

```diff
@@ -410,4 +410,8 @@ package object test extends AssertionVariants {
   private final def reassociate[A, B, C, D, E](f: (A, B, C, D) => E): ((((A, B), C), D)) => E = {
     case (((a, b), c), d) => f(a, b, c, d)
   }
+
+  implicit class ZioOfTestResultOps[R, E](val res: ZIO[R, E, TestResult]) {
```

Comment:
Can we take this out? We have an open issue for the same problem you're dealing with in #1888 and do want to do something about it, but if we do something want to support all the operators and do so in a way that works for all combinations of `assert` and `assertM`. Not sure if implicit syntax is the right way to do this. If we make assertions effectual as I am working on in #2182  this problem will solve itself. In the meantime you can do something like:

```scala
  for {
    partial <- assertM(partialParse, succeeds(equalTo(List.empty[Int])))
    full    <- assertM(fullParse, succeeds(equalTo(List(123, 4))))
  } yield partial && full
```

```scala
  for {
    partial <- partialParse
    full    <- fullParse
  } yield assert(partial, succeeds(equalTo(List.empty[Int]))) &&
    assert(full, succeeds(equalTo(List(123, 4))))
```

Reply:
I've used this operator a lot in this test so I moved it into the helper object for the Spec, for now

Reply:
Good idea.